libgphoto2/gphoto2-abilities-list.c

```diff
@@ -212,10 +213,12 @@ gp_abilities_list_load_dir (CameraAbilitiesList *list, const char *dir,
 	if (1) { /* a new block in which we can define a temporary variable */
 		foreach_data_t foreach_data = { NULL, GP_OK };
 		foreach_data.list = flist;
+		gpi_libltdl_lock();
```

Comment:
Looking at frequent locks/unlocks throughout the function, I wonder if it would've been easier to

1. Rename the unmodified function to `gp_abilities_list_load_dir_impl`.
2. Add a function `gp_abilities_list_load_dir` that locks libltdl, invokes `gp_abilities_list_load_dir_impl`, unlocks libltdl and then returns the return code.

Reply:
Same for other functions that are dedicated to loading libs, and consist for ~90% of libltdl calls anyway.

Reply:
@ndim what do you think? seems sensible

Reply:
While I'm working on other parts, let's do this anyway ^

Reply:
i did this for this function.